packages/common/src/hooks/chats/useAudiusLinkResolver.ts

```diff
@@ -177,64 +200,47 @@ export const useAudiusLinkResolver = ({
       cursorPosition: number
       textBeforeCursor: string
     }) => {
-      const matched = Object.keys(humanToData).find((i) =>
-        textBeforeCursor.endsWith(i)
-      )
+      const sortedStrings = getSortedMatchStrings()
+      const matched = sortedStrings.find((i) => textBeforeCursor.endsWith(i))
       if (matched) {
-        return (value: string) =>
-          value.slice(0, cursorPosition - matched.length) +
-          value.slice(cursorPosition)
+        const newCursorPosition = cursorPosition - matched.length
+        return {
+          editValue: (value: string) =>
+            value.slice(0, newCursorPosition) + value.slice(cursorPosition),
+          newCursorPosition
+        }
+      }
+      return {
+        editValue: null,
+        newCursorPosition: cursorPosition
       }
-      return null
     },
-    [humanToData]
+    [getSortedMatchStrings]
   )
 
   const getMatches = useCallback(
     (value: string) => {
-      const regexString = Object.keys(humanToData).join('|')
-      const regex = regexString ? new RegExp(regexString, 'gi') : null
-      if (regex) {
-        return value.matchAll(regex)
-      }
-      return null
+      const regexString = getSortedRegex()
+      const regex = regexString ? new RegExp(regexString, 'g') : null
+
+      return regex
+        ? Array.from(value.matchAll(regex)).map((match) => ({
+            type: 'match',
+            text: match[0],
+            index: match.index,
+            link: humanToData[match[0]].link
+          }))
+        : null
     },
-    [humanToData]
+    [getSortedRegex, humanToData]
   )
 
-  /**
-   * Resets any found track id / collection id
-   */
-  const clearLinks = useCallback(() => {
-    setTrackId(null)
-    setCollectionId(null)
-  }, [setTrackId, setCollectionId])
-
-  /**
-   * Updates the track id and/or collection id state if found in matching text
-   */
-  useEffect(() => {
-    for (const [human, { data }] of Object.entries(humanToData)) {
-      if (value.includes(human)) {
-        if (instanceOfTrack(data)) {
-          setTrackId(decodeHashId(data.id))
-        } else if (instanceOfPlaylist(data)) {
-          setCollectionId(decodeHashId(data.id))
-        }
-        return
-      }
-    }
-    setTrackId(null)
-    setCollectionId(null)
-  }, [trackId, humanToData, value])
```

Comment:
No context here; curious why these were removed?

Reply:
the `trackId` and `collectionId` state was very specific to DMs so replaced it with a more general `linkEntities` state